modules/primo/includes/primo.search.inc

```diff
@@ -283,12 +283,24 @@ function primo_search_sort_options() {
  *   Items similar to the query.
  */
 function primo_search_autocomplete($query) {
-  // TODO: Only search on titles, and do something to speed up the query.
-  $results = ting_start_query()->setCount(10)->setFullTextQuery($query)->execute();
-
   $suggestions = [];
-  foreach ($results->getTingEntityCollections() as $collection) {
-    $suggestions[$collection->getTitle()] = $collection->getTitle();
+
+  try {
+    $results = primo_client()
+      ->search('swtitle,begins_with,' . urlencode($query), 1, 10)
+      ->getDocuments();
+  }
+  catch (TransferException $e) {
+    watchdog('primo', '@error_code: @message',
+      ['@error_code' => $e->getCode(), '@message' => $e->getMessage()],
+      WATCHDOG_ERROR
+    );
```

Comment:
Do you know [`watchdog_exception`](https://api.drupal.org/api/drupal/includes%21bootstrap.inc/function/watchdog_exception/7.x)? It is actually a nice way to handle this kind of logging.

Reply:
I totally forgot it existed! Thanks. :)